diff --git a/main.c b/main.c
index f78ca4c..9c18d6a 100644
--- a/main.c
+++ b/main.c
@@ -43,6 +43,7 @@ static struct option long_options[] = {
     {"no-wrap", 0, NULL, 'w'},
     {"no-banner", 0, NULL, 'b'},
     {"help", 0, NULL, 'h'},
+    {"no-greeting", 0, NULL, 'g'},
     {"directory", 1, NULL, 'd'},
     {0, 0, 0, 0}
 };
@@ -80,7 +81,7 @@ int main(int argc, char **argv)
     directory_set = 0;
 
     while(1) {
-	if((c = getopt_long(argc, argv, "hpwbd:", long_options,
+	if((c = getopt_long(argc, argv, "hpwbgd:", long_options,
 			    &option_index)) == -1)
 	    break;
 	switch(c) {
@@ -97,6 +98,9 @@ int main(int argc, char **argv)
 	case 'b':
 	    megahal_setnobanner();
 	    break;
+    case 'g':
+        megahal_setnogreeting();
+        break;
 	case 'h':
 	    usage();
 	    return 0;
diff --git a/megahal.c b/megahal.c
index b67767e..a396d13 100644
--- a/megahal.c
+++ b/megahal.c
@@ -224,6 +224,7 @@ static bool speech=FALSE;
 static bool quiet=FALSE;
 static bool nowrap=FALSE;
 static bool nobanner=FALSE;
+static bool nogreeting=FALSE;
 
 static char *errorfilename = "megahal.log";
 static char *statusfilename = "megahal.txt";
@@ -380,6 +381,10 @@ void megahal_setnobanner (void)
 {
     nobanner = TRUE;
 }
+void megahal_setnogreeting (void)
+{
+    nogreeting = TRUE;
+}
 
 void megahal_seterrorfile(char *filename)
 {
@@ -497,9 +502,13 @@ char *megahal_initial_greeting(void)
 {
     char *output;
 
-    make_greeting(greets);
-    output = generate_reply(model, greets);
-    return output;
+    if (!nogreeting) {
+        make_greeting(greets);
+        output = generate_reply(model, greets);
+        return output;
+    } else {
+        return "";
+    }
 }
 
 /*
diff --git a/megahal.h b/megahal.h
index 6da8032..2ac2409 100644
--- a/megahal.h
+++ b/megahal.h
@@ -113,6 +113,7 @@ extern void srand48(long);
 void megahal_setnoprompt (void);
 void megahal_setnowrap (void);
 void megahal_setnobanner (void);
+void megahal_setnogreeting (void);
 
 void megahal_seterrorfile(char *filename);
 void megahal_setstatusfile(char *filename);
